[New product] Add IBM i operating system (AS400)

adriens opened the pull request:
# :information_source:  About

The aim of this PR is to add [IBM i operating system](https://www.ibm.com/products/ibm-i) to `endoflife.date`.

I have mostly used hte following resources:

- https://www.ibm.com/support/pages/ibm-i-release-support
- https://www.ibm.com/support/pages/release-life-cycle

# :grey_question: Question & Feedback

Please let me know if you think we sould merge releases, like : 

- `7.5`, ... `7.1` into **7**
- Same question for `5.x`
- `4.x`
- `3.x`

adriens committed:
[New product] Add IBM i operating system

adriens committed:
IMB-i > Add unix tag

adriens committed:
Update ibm-i.md > typo

cf https://github.com/endoflife-date/endoflife.date/pull/4063#issuecomment-1804593390

adriens committed:
Update ibm-i.md > typos

cf https://github.com/endoflife-date/endoflife.date/pull/4063#issuecomment-1805244084

BiNZGi commented on products/ibm-i.md:
```diff
@@ -0,0 +1,167 @@
+-   /i-series
+-   /ibm-iseries
+-   /ibmi
+-   /as400
+-   /os400
```
Please add these two "names" also in text. Our search function does not find as400 or os400 but I think we should hit this page if a user is looking for that. WDYT?

adriens replied:
Hi @BiNZGi , I've commited all your suggestions and is curretly trying to add `extendedSupport`

BiNZGi commented on products/ibm-i.md:
```diff
@@ -0,0 +1,167 @@
+    latest: "5.4"
+    latestReleaseDate: 2006-02-14
+    
+
```
Only one line:
```suggestion

```

BiNZGi commented on products/ibm-i.md:
```diff
@@ -0,0 +1,167 @@
+
+-   releaseCycle: "4.4"
+    releaseDate: 1999-05-21
+    eol: 2001-05-30
```
https://www.ibm.com/support/pages/release-life-cycle says:
```suggestion
    eol: 2001-05-31
```

BiNZGi commented on products/ibm-i.md:
```diff
@@ -0,0 +1,167 @@
+
+_Note 4_: IBM i 7.2 Service Extension Offering started on May 1, 2021. For details regarding support coverage for the various products in the IBM i portfolio, see Service Extension for IBM i 7.3, 7.2 and 7.1.
+
+_Note 5_: IBM i 7.3 Service Extension Offering starts on October 1, 2023. For details regarding support coverage for the various products in the IBM i portfolio, see Service Extension for IBM i 7.3, 7.2 and 7.1.
```
The notes are not reference, I would remove the numbering. I think the user is looking for the latest version at first, I would reverse the order:
```suggestion
End of program support date will be announced with at least 12 months notice prior to the effective termination date.

IBM i 7.3 Service Extension Offering starts on October 1, 2023. For details regarding support coverage for the various products in the IBM i portfolio, see Service Extension for IBM i 7.3, 7.2 and 7.1.

IBM i 7.2 Service Extension Offering started on May 1, 2021. For details regarding support coverage for the various products in the IBM i portfolio, see Service Extension for IBM i 7.3, 7.2 and 7.1.

IBM i 7.1 extended Service Extension started on May 1, 2021. The duration of IBM i 7.1 extended Service Extension is dependent on the Power hardware generation. The extended Service Extension coverage is usage and known defect support, and there some exceptions of products and functions which are not supported. For notes regarding extended Service Extension duration and support coverage, see Service Extension for IBM i 7.3, 7.2 and 7.1.

For 6.1, includes both Machine Code Level V6R1M0 and V6R1M1.
```
WDYT?

BiNZGi commented on products/ibm-i.md:
```diff
@@ -0,0 +1,167 @@
+    eol: 1997-05-31
+    latest: "3.0.5"
+    latestReleaseDate: 1994-06-03
+
```
On https://www.ibm.com/support/pages/release-life-cycle are also cycles for 2.3, 2.2, and 2.1:
```suggestion

-   releaseCycle: "2.3"
    releaseDate: 1993-12-17
    eol: 1996-05-31
    latest: "2.3"
    latestReleaseDate: 1993-12-17
    
-   releaseCycle: "2.2"
    releaseDate: 1992-09-18
    eol: 1995-03-31
    latest: "2.2"
    latestReleaseDate: 1992-09-18

-   releaseCycle: "2.1"
    releaseDate: 1991-05-24
    eol: 1994-06-30
    latest: "2.1.1"
    latestReleaseDate: 1992-03-06
    
```
WDYT?

BiNZGi reviewed (commented):
Overall looks good, I have some suggestions in separate comments.

adriens committed:
Update ibm-i.md > Add extendedSupport

cf https://github.com/endoflife-date/endoflife.date/pull/4063#issuecomment-1819623885

adriens committed:
Update products/ibm-i.md

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

adriens committed:
Update products/ibm-i.md

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

adriens committed:
Update products/ibm-i.md

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

adriens committed:
Update products/ibm-i.md

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

adriens committed:
Update ibm-i.md

adriens committed:
Update ibm-i.md

adriens committed:
Update ibm-i.md

BiNZGi commented on products/ibm-i.md:
```diff
@@ -0,0 +1,208 @@
+    
+---
+
+> [IBM i](https://www.ibm.com/products/ibm-i) is a fully integrated operating system, meaning the database, middleware, security, runtime and hypervisor are integrated into the stack and licensed as one solution.
```
I tought about adding a bit of the historical names:
```suggestion
> [IBM i](https://www.ibm.com/products/ibm-i) is a fully integrated operating system, meaning the database, middleware, security, runtime and hypervisor are integrated into the stack and licensed as one solution. It was originally released in 1988 as OS/400, as the sole operating system of the IBM AS/400 line of systems. It was renamed to i5/OS in 2004, before being renamed a second time to IBM i in 2008.
```
WDYT?

BiNZGi commented on products/ibm-i.md:
```diff
@@ -0,0 +1,208 @@
+    extendedSupport: false
+
+-   releaseCycle: "6.1"
+    releaseDate: 2008-04-23
```
According to https://www.ibm.com/support/pages/release-life-cycle:
```suggestion
    releaseDate: 2008-03-21
```

adriens committed:
Update products/ibm-i.md

🙏

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

adriens committed:
Update products/ibm-i.md

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

BiNZGi reviewed (approved):
All good now :+1: nice work @adriens :tada: -> merge
